src/components/probe/prober/http/response-cache.ts

```diff
@@ -0,0 +1,98 @@
+/**********************************************************************************
+ * MIT License                                                                    *
+ *                                                                                *
+ * Copyright (c) 2021 Hyperjump Technology                                        *
+ *                                                                                *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy   *
+ * of this software and associated documentation files (the "Software"), to deal  *
+ * in the Software without restriction, including without limitation the rights   *
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      *
+ * copies of the Software, and to permit persons to whom the Software is          *
+ * furnished to do so, subject to the following conditions:                       *
+ *                                                                                *
+ * The above copyright notice and this permission notice shall be included in all *
+ * copies or substantial portions of the Software.                                *
+ *                                                                                *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    *
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  *
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
+ * SOFTWARE.                                                                      *
+ **********************************************************************************/
+
+import { getContext } from '../../../../context'
+import { ProbeRequestResponse } from 'src/interfaces/request'
+import { log } from '../../../../utils/pino'
+import { serialize } from 'v8'
+/**
+ * Simple implementation for caching Monika HTTP responses.
+ * With default total cache size limited to 50 MB.
+ * And default time-to-live for each cache entries is 30s.
+ *
+ * About cache size 50 MB limit :
+ * Assuming a typical web page response is around 500 KB
+ * This cache can fit around 100 entries of web pages
+ * A typical response with empty body is around 300 bytes of headers
+ * That means, this cache can fit around 160K entries of empty body responses
+ */
+const DEFAULT_CACHE_LIMIT = 50_000_000 // 50 MB in bytes
+const DEFAULT_TIME_TO_LIVE = 30_000 // 30s in ms
```

Comment:
Should the cache limit and TTL configurable through flags?

Reply:
I am thinking in the real case scenario where probing may be 5 minutes per probe, should the TTL be longer?

Reply:
updated